test/extended/util/client.go

```diff
@@ -191,9 +192,9 @@ func (c *CLI) Username() string {
 
 // AsAdmin changes current config file path to the admin config.
 func (c *CLI) AsAdmin() *CLI {
-	nc := *c
-	nc.configPath = c.adminConfigPath
-	return &nc
+	c.currentConfigPath = c.adminConfigPath
```

Comment:
in case of multiple `CLI` modifications within a test, would it make sense to protect all these changes with mutexes?

Reply:
I'd leave it up to consumers to deal with that kind of changes. I can probably add a comment to `NewCLI` about not having these kind of guarantees.